6MI0800226/week2.cpp

```diff
@@ -0,0 +1,160 @@
+#include <iostream>
+#include <fstream>
+#include <assert.h>
+ 
+const int BUFF_SIZE = 1024;
+void saveResultsInFile(int num1, int num2, const char location[BUFF_SIZE])
+{
+    std::ofstream file(location);
+ 
+    if (!file.is_open())
+        return;
+ 
+    file << (num1 + num2) << std::endl << (num1 * num2);
+}
+int findDiffInFile(const char location[BUFF_SIZE])
+{
+    int num1, num2;
+    std::ifstream file(location);
+ 
+    if (!file.is_open())
+        return -1;
+ 
+    file >> num1 >> num2;
+    return num1 - num2;
+}
+ 
+int sizeOfFile(const char location[BUFF_SIZE])
+{
+    int size = 0;
+    std::ifstream file(location);
+ 
+    if (!file.is_open())
+        return -1;
+ 
+    while (!file.eof())
+    {
+        char buffM[BUFF_SIZE];
+        char* buff = buffM;
```

Comment:
That's rather strange decision - having the `buff` variable instead of iterating over `buffM` :thinking: Anyways - if you have a file that contains a line longer than `BUFF_SIZE` - you're going to get into trouble. Using the [get](https://cplusplus.com/reference/istream/istream/get/) that takes also max number of chars to read is the better option. You can use [gcount](https://cplusplus.com/reference/istream/istream/gcount/) afterwards to identify how many symbols were in fact read.